packages/@aws-cdk/aws-apigatewayv2/lib/http/api-mapping.ts

```diff
@@ -0,0 +1,92 @@
+import { Construct, Resource } from '@aws-cdk/core';
+import { CfnApiMapping, CfnApiMappingProps } from '../apigatewayv2.generated';
+import { IApiMapping, IDomainName } from '../common';
+import { IHttpApi } from '../http/api';
+import { IHttpStage } from './stage';
+
+/**
+ * Properties used to create the HttpApiMapping resource
+ */
+export interface HttpApiMappingProps {
+  /**
+   * Api mapping name
+   * @default -  logical id
+   */
+  readonly apiMappingName?: string;
+
+  /**
+   * Api mapping key. Specify `/` for the root path of the domain.
+   */
+  readonly apiMappingKey: string;
```

Comment:
Sorry, I know my previous comment mentioned about making this mandatory but I think this is better - 

```suggestion
  /**
   * Api mapping key. The path where this stage should be mapped to on the domain
   * @default '/'
   */
  readonly apiMappingKey?: string;
```